File: tests/CompileTests/OpenMP_tests/single.c
/*
 loop scheduling
*/
#include <stdio.h>
#ifdef _OPENMP
#include <omp.h>
#endif 

int main(void)
{
  int i = 100;

#pragma omp parallel
  {
#pragma omp single firstprivate(i)
    {
      i += omp_get_num_threads();
      printf ("i= %d \n",i);
    }
  }
}

File: tests/CompileTests/OpenMP_tests/fortran/single.f
C***************************************************
C Liao, 1/20/2010
C***************************************************
      PROGRAM HELLO
      include 'omp_lib.h'
!$OMP PARALLEL
!$OMP SINGLE
      print *, 'Running on ', omp_get_num_threads(), 'threads'
!$omp END SINGLE
      PRINT *, 'Hello World!'
c$omp end parallel 
      PRINT *, 'the end'
      END
